packages/flutter/test/material/data_table_test.dart

```diff
@@ -1323,13 +1332,25 @@ void main() {
       );
     }
 
-    Checkbox lastCheckbox() {
-      return tester.widgetList<Checkbox>(find.byType(Checkbox)).last;
+    RenderObject lastCheckboxPainter() {
+      return tester.renderObject(
+        find.descendant(of: find.byType(Checkbox), matching: find.byType(CustomPaint)).last,
+      );
+    }
+
+    PaintPattern paintsPath(PaintingStyle style, Color color) {
+      return paints..something((Symbol method, List<dynamic> arguments) {
+        if (method != #drawPath)
+          return false;
+        final Paint paint = arguments[1] as Paint;
+        return paint.style == PaintingStyle.fill && paint.color == color;
+      });
     }
 
     await tester.pumpWidget(buildTable());
-    expect(lastCheckbox().activeColor, _themeData.colorScheme.primary);
-    expect(lastCheckbox().checkColor, _themeData.colorScheme.onPrimary);
+    await tester.pump();
+    expect(lastCheckboxPainter(), paintsPath(PaintingStyle.fill, fillColor));
+    expect(lastCheckboxPainter(), paintsPath(PaintingStyle.stroke, checkColor));
```

Comment:
IMO it would be cleaner if this is tested the same way it's done in `checkbox_theme_test.dart`. WDYT?

https://github.com/flutter/flutter/blob/d719463489218f9de5f7b8b2237cb256f7617e9f/packages/flutter/test/material/checkbox_theme_test.dart#L138-L142

```suggestion
    expect(
        Material.of(tester.element(find.byType(Checkbox).last)),
        paints
          ..path()
          ..path(color: fillColor)
          ..path(color: checkColor)
    );
```

Reply:
Sure! Will do momentarily. Thanks for the review 👍🏻